src/Aspire.Hosting.Yarp/ConfigurationBuilder/YarpCluster.cs

```diff
@@ -53,19 +62,36 @@ private static string BuildEndpointUri(IResourceWithServiceDiscovery resource)
     {
         var resourceName = resource.Name;
 
+        // NOTE: This should likely fallback to other endpoints with HTTP or HTTPS schemes in cases where they don't
+        //       have the default names.
         var httpsEndpoint = resource.GetEndpoint("https");
         var httpEndpoint = resource.GetEndpoint("http");
 
         var scheme = (httpsEndpoint.Exists, httpEndpoint.Exists) switch
         {
-            (true, true)  => "https+http",
+            (true, true) => "https+http",
             (true, false) => "https",
             (false, true) => "http",
             _ => throw new ArgumentException("Cannot find a http or https endpoint for this resource.", nameof(resource))
         };
 
         return $"{scheme}://{resourceName}";
     }
+
+    private static string GetAddressFromExternalService(ExternalServiceResource externalService)
+    {
+        if (externalService.Uri is not null)
+        {
+            return externalService.Uri.ToString();
+        }
+        if (externalService.UrlParameter is not null)
+        {
+            // BUG: If we're in publish mode we shouldn't be accessing the parameter value.
+            return externalService.UrlParameter.Value;
```

Comment:
@davidfowl what's your feeling on what we should do here? Throw in the `AddCluster` method in publish mode? Or are we going to address publishability of Yarp in general separately?

Reply:
We can make this work when @benjaminpetit does his next PR